### PR TITLE
Improve release-notes code reusability

### PR DESCRIPTION
**Pull request by machadovilaca**
I was looking to add a tool on kubevirt/must-gather to generate release notes
and noticed most of the code would be very similar to the one used to generate
the release notes in this repository. My suggestion would be splitting the code
that handles all the git related operations to a different package as I did in
this PR. Other option could be moving the git package here created to 
kubevirt/controller-lifecycle-operator-sdk in the 'pkg/sdk' directory and then
import it both in the HCO and kubevirt/must-gather repositories release-notes.

Signed-off-by: João Vilaça <jvilaca@redhat.com>

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
None
```

### CODE REVIEW

**Review by nunnatsa — COMMENTED** [review]
General comment: did you consider to use a go-git driver package instead of calling another process? e.g. https://git-scm.com/book/en/v2/Appendix-B%3A-Embedding-Git-in-your-Applications-go-git?

**Comment by machadovilaca** [comment]
> General comment: did you consider to use a go-git driver package instead of calling another process? e.g. https://git-scm.com/book/en/v2/Appendix-B%3A-Embedding-Git-in-your-Applications-go-git?

This PR is just a reorganization of files. But looks like a good approach that would clean and simplify a lot of the code. Will try

File: tools/release-notes/release-notes.go [hunk 1]
```diff
@@ -56,14 +57,14 @@ func (r *releaseData) writeHcoChanges(span string) error {
 	return nil
 }
 
-func (p *project) writeOtherChangesIfVersionUpdated(f *os.File) error {
-	span := fmt.Sprintf("%s..%s", p.previousTag, p.currentTag)
-	releaseNotes, err := p.gitGetReleaseNotes(span)
+func writeOtherChangesIfVersionUpdated(g *git.Project, f *os.File) error {
```
Comment (nunnatsa) [inline tools/release-notes/release-notes.go:60]: Please consider to use `io.Writer` instead of `*os.File`

This will

Comment (machadovilaca) [reply tools/release-notes/release-notes.go]: done

**Review by nunnatsa — COMMENTED** [review]
Sorry for the late response. 

I added two small comments about readability.

File: tools/release-notes/git/git.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+func (g *Project) releaseNoteFromMergeCommit(releaseNotes []string, line string) []string {
+	pr := strings.Split(line, " ")
+
+	num, err := strconv.Atoi(strings.TrimPrefix(pr[3], "#"))
+	if err != nil {
+		return releaseNotes
+	}
+	return g.readReleaseNote(releaseNotes, num)
+}
```
[the remark this answers is not in the document]
Comment (machadovilaca) [reply tools/release-notes/git/git.go]: updated

File: tools/release-notes/git/git.go [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+			releaseNotes = g.releaseNoteFromMergeCommit(releaseNotes, commit.Message)
+		}
+	}
+
+	return releaseNotes, nil
+}
+
+func (g *Project) releaseNoteFromSquashCommit(releaseNotes []string, matches []string) []string {
+	for _, match := range matches {
+		num, err := strconv.Atoi(match[2 : len(match)-1])
```
[the remark this answers is not in the document]
Comment (machadovilaca) [reply tools/release-notes/git/git.go]: added

**Pull request merged**